wled00/FX_fcn.cpp

```diff
@@ -462,6 +458,110 @@ CRGBPalette16 &Segment::currentPalette(CRGBPalette16 &targetPalette, uint8_t pal
   return targetPalette;
 }
 
+CRGBPalette16 Segment::generateRandomPalette(){
```

Comment:
As generating palettes has no connection with segment class I'd recommend moving this function out of Segment class and make it global *color utility* function (`color.cpp`).